### PR TITLE
Updated for new Vimeo API response format, added best available quality

### PR DESCRIPTION
**Pull request by johnrogers**
Fix for issue #24.

I've used the code provided by @Chris59160 and tidied it up a bit.

I found myself requiring the best available video quality, so I've added code to facilitate this, too. I feel this is a common requirement for users of this code.

### CODE REVIEW

**Comment by jvanderwee** [comment]
It'd be great if this could be merged and released!

**Comment by lilfaf** [comment]
@johnrogers thank you for your contribution! Allow users to fetch the best available quality is a really great idea. 
However, your patch involves a regression, we lose the ability to auto fallback to lower available quality. 
Vimeo also added new HD video presets so we should update `YTVimeoVideoQuality`. I'm trying to figure out a way to make the transition without breaking backward compatibility.
I've pushed a quick fix on the hotfix/vimeoapi branch.

**Comment by jvanderwee** [comment]
Hero

**Comment by tjmaynes** [comment]
:+1:

File: YTVimeoExtractor/YTVimeoExtractor.m [hunk 1]
```diff
@@ -189,57 +189,59 @@ - (void)URLSession:(NSURLSession *)session dataTask:(NSURLSessionDataTask *)data
...
+	NSArray *videoArray = [jsonData valueForKeyPath:@"request.files.progressive"];
+	if (!videoArray) {
+		[self extractorFailedWithMessage:@"Unsupported video codec" errorCode:YTVimeoExtractorErrorUnsupportedCodec];
+		return;
+	}
+	
+	NSDictionary *videoInfos;
+	if (videoArray.count > 0) {
+		if (self.quality == YTVimeoVideoQualityBestAvailable) {
+			videoInfos = [videoArray lastObject];
```
Comment (pixelkind) [inline YTVimeoExtractor/YTVimeoExtractor.m]: It is not guaranteed that the videoInfos array is always in the right order so you have to sort them to be sure to pick the best solution. For an easy way to do so, have a look at my pull request.

**Comment by lilfaf** [comment]
Sorry all for the long waiting, I couldn't find some time to make the few changes the lib needed. All Issues opened issues should be fixed.
I depreciated the old `YTVimeoVideoQuality` options and introduced some more generic names (inspired by [XCDYouTubeKit](https://github.com/0xced/XCDYouTubeKit)). It should be easier to name future qualities if vimeo decides to support more.
@johnrogers @pixelkind @SoneeJohn Thank you for all your contributions ! It helped me a lot.

**Comment by SoneeJohn** [comment]
@lilfaf It was a pleasure to help :)

**Pull request closed without merging**